### ISSUE TITLE
`pub type NewName = S` does not work for nullary struct `S`

### ISSUE THREAD

**Issue by pnkfelix** [rust-lang/rust#22259]
`pub type NewName = S` does not work for nullary struct `S`

Sample program:

```rust
mod a {
    pub struct Aliased1 { pub x: u32 }
    pub type Public2 = Aliased1;

    pub struct Unaliased3;

    pub struct Aliased4;
    pub type Public5 = Aliased4;
}
fn main() {
    use a;
    let _p = a::Public2 { x: 10 };
    let _q = a::Unaliased3;
    let _r = a::Public5;
}
```

produces this in the playpen:

```
<anon>:14:14: 14:24 error: unresolved name `a::Public5`
<anon>:14     let _r = a::Public5;
                       ^~~~~~~~~~
error: aborting due to previous error
playpen: application terminated with error code 101
```

**Comment by P1start** [comment]
This seems intentional to me. The constructor for unit structs is just a value (in the value namespace), but `type` creates aliases in the type namespace. As a workaround, one can use a `const`:

```rust
mod a {
    pub struct Aliased1 { pub x: u32 }
    pub type Public2 = Aliased1;

    pub struct Unaliased3;

    pub struct Aliased4;
    pub type Public5 = Aliased4;
    #[allow(non_upper_case_globals)]
    pub const Public5: Aliased4 = Aliased4;
}
fn main() {
    use a;
    let _p = a::Public2 { x: 10 };
    let _q = a::Unaliased3;
    let _r = a::Public5;
}
```

Also, I think this is a dupe of the (closed) issue #17422.

**Comment by pnkfelix** [comment]
ah of course; I should I have seen this reasoning
